extensions/wikia/ContentReview/controllers/ContentReviewDiffPage.class.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+namespace Wikia\ContentReview;
+
+use Wikia\ContentReview\Models\ReviewModel;
+
+class ContentReviewDiffPage extends \ContextSource {
+
+	const CONTENT_REVIEW_TOOLBAR_TEMPLATE_PATH = 'extensions/wikia/ContentReview/templates/ContentReviewToolbar.mustache';
+
+	private
+		$escalated,
+		$pageId,
+		$revisionId,
+		$status,
+		$title,
+		$toolbarHtml,
+		$wikiId;
+
+	public function __construct( \Title $title ) {
+		global $wgCityId;
+
+		$this->wikiId = $wgCityId;
+		$this->title = $title;
+		$this->pageId = $title->getArticleID();
+		$this->revisionId = $this->getCurrentlyReviewedRevisionId();
+	}
+
+	/**
+	 * Tries to initiate and render the toolbar. Triggers checks if for conditions
+	 * for displaying the toolbar being fulfilled. If not - it returns false.
+	 * Sets toolbarHtml, prepares assets and returns true otherwise.
+	 * @return bool
+	 */
+	public function initReviewerToolbar() {
```

Comment:
I think it might be better to move the rendering of the HTML and adding of assets to the output to the `addToolbarToOutput` method (and make `$toolbarHtml` a local var to that method as it isn't used anywhere else) and then change the hook to just use:

```php
if ( $diffPage->shouldDisplayToolbar() ) {
	$diffPage->addToolbarToOutput( $output );
}
```

As I'm not sure of the reasoning for initiating that data that is only relevant to modifying the output in the check for whether or not to add the output.